holoviews/plotting/bokeh/renderer.py

```diff
@@ -212,6 +216,11 @@ def _figure_data(self, plot, fmt='html', doc=None, **kwargs):
         for m in model.references():
             m._document = None
         doc.add_root(model)
+
+        if self.theme:
+            from bokeh.io import curdoc
+            curdoc().theme = self.theme
```

Comment:
Seems like this should be ``doc.theme`` not ``curdoc().theme``, but maybe I'm confused?

Reply:
Well spotted! I had this comment in my code:

```
# Using curdoc because of conditional in embed.py _ModelInEmptyDocument
```

I'll try using ``doc`` again but I suspect it might not work for this reason - I forgot the details exactly. If ``doc`` does work, I also prefer using that.

Reply:
I can also restore that comment if that helps.

Reply:
Right, I just checked and using ``doc`` does not work.  I'm not sure which conditional I am referring to, maybe this one https://github.com/bokeh/bokeh/blob/master/bokeh/embed/notebook.py#L121?

Only in that branch does it look like the theme is transferred from one document to another in the way you expect from the bokeh API (setting the theme on the document). Perhaps there is a way to pass ``apply_theme`` into the ``_ModelInEmptyDocument`` context manager somehow but I would not know what that is.

Reply:
Okay, I'll have a poke around as well. Want to at least make sure it works when a user supplies a custom Document, that's not equivalent to curdoc().

Reply:
Good idea!